gadgetchains/Smarty/FD/1/gadgets.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+abstract class Smarty_CacheResource
+{}
+
+class Smarty_Internal_CacheResource_File extends Smarty_CacheResource
+{
+    public function releaseLock(Smarty $smarty, Smarty_Template_Cached $cached)
+    {
+        $cached->is_locked = false;
+        unlink($cached->lock_id);
+    }
+}
+
+class Smarty_Template_Cached
+{
+    public $lock_id = null;
+    public $is_locked = true;
+
+    public function __construct()
+    {
+        $this->handler = new Smarty_Internal_CacheResource_File();
+        $this->lock_id = '';
+    }
+
+    public function setlock($lock_id){
+        if($lock_id){
+            $this->lock_id = $lock_id;
+        }
+    }
+}
+
+class Smarty_Internal_TemplateBase
+{}
+
+class Smarty extends Smarty_Internal_TemplateBase{
```

Comment:
The code style for PHPGGC is to use a new line before opening a block, like so:
```
class Smarty extends Smarty_Internal_TemplateBase
{
```

Reply:
Done!